docs/userguide/canvas.rst

```diff
@@ -1122,3 +1122,53 @@ of one:
 
 This means that the first task will have a countdown of one second, the second
 task a countdown of two seconds, and so on.
+
+Stamping
+=======
+The goal of the Stamping API is to give an ability to mark signature and it components for a debug purposes. For example,
+
+.. code-block:: pycon
+
+    >>> sig1 = add.si(2, 2)
+    >>> sig1_res = sig1.freeze()
+    >>> g = group(sig1, add.si(3, 3))
+    >>> g.stamp(stamp='your_custom_stamp')
+    >>> res = g1.apply_async()
+    >>> res.get(timeout=TIMEOUT)
+    [4, 6]
+    >>> sig1_res._get_task_meta()['stamp']
+    ['your_custom_stamp']
+
+will initialize a group ``g`` and mark its components with stamp ``your_custom_stamp``.
+
+For the usage of stamping API to use this feature, you need to set up the :ref:`configuration <configuration:result_extended>` ``CELERY_RESULT_EXTENDED = True``
```

Comment:
There's a custom directive for referencing settings. See https://github.com/celery/celery/blob/c506f45926b40c041bfbe5147e3a3e59a9751435/docs/userguide/optimizing.rst#L58 for example.